libraries/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts

```diff
@@ -786,11 +808,15 @@ export class PhasedScriptAction extends BaseScriptAction<IPhasedCommandConfig> {
 
     // Loop until Ctrl+C
     // eslint-disable-next-line no-constant-condition
-    while (true) {
+    while (!abortSignal.aborted) {
       // On the initial invocation, this promise will return immediately with the full set of projects
       const { changedProjects, inputsSnapshot: state } =
         await projectWatcher.waitForChangeAsync(onWaitingForChanges);
 
+      if (abortSignal.aborted) {
```

Comment:
Should this and the abort signal be checked? Seems odd if the abort signal doesn't abort while waiting for changes

Reply:
My initial implementation causes `waitForChangeAsync` to return a result when aborted, even if the result may be that 0 projects have changed, so the separate check is here to know if we are in that path.